Domain management revamp: update domain list column styles in sidebar

Imran92 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/96554

## Proposed Changes
Updates the looks of the left side of the Domains preview

* Fix the font size, spacing and styling of the Tab title ("Domains")
* Fix the look and style of the button to add domains
* Fix search bar
* Fix spacings
* Remove column names when showing domains only in sidebar
* Fix hiding left column if size of tab view and smaller
* Fix paddings of preview pane in mobile view
* Fixes scrollbar showing up in tab

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* To revamp domain management

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Go to wordpress.com/domains/manage
- Browse around, check inner functionalities, make sure you see no issues in how the table looks and works
- Open a site's wp-admin area, go to Upgrades -> Emails, browse around and make sure everything works as before
- Also click on a few domains to make sure everything is working as before.
- Now enable the feature flag by entering window.sessionStorage.setItem('flags', 'calypso/all-domain-management') in the - browser console. Reload the page
- Now click on different domains. They should open in the new layout now.
- Check that the left side of the new layout (the domains list) matches the design.
- When you open wordpress.com/sites and click a site, the left side panel opens. The domains side panel should also match the sites side panel

| Before | After |
|--------|--------|
| <img width="1545" alt="Screenshot 2024-12-07 at 1 50 39 AM" src="https://github.com/user-attachments/assets/2e59600d-e2d3-4242-9997-62124c360e44"> | <img width="1619" alt="Screenshot 2024-12-07 at 3 06 44 AM" src="https://github.com/user-attachments/assets/7eb3a716-49e0-48fd-b79a-87cc9044dff5"> |

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Imran92 committed:
Fix content padding in mobile view

Imran92 committed:
Fix split button primary style for sidebar mode for flyout view

Imran92 committed:
Add a comment to common function

Imran92 committed:
Hide column headers in sidebar view of site list

Imran92 committed:
Import stylesheet

Imran92 committed:
Fix left title and button above domain list

Imran92 committed:
Fix domain table item, search bar styles and spacings

Imran92 committed:
Fix title spacings

m1r0 commented on client/my-sites/domains/domain-management/domain-overview-pane/style.scss:
```diff
@@ -0,0 +1,7 @@
+@media (max-width: 960px) {
```
What do you think of using the [media query mixins](https://github.com/Automattic/wp-calypso/blob/trunk/docs/coding-guidelines/css.md#media-queries) here instead?

Imran92 replied:
Done it here https://github.com/Automattic/wp-calypso/pull/97193/commits/ddd028a0cb3a9c56036dda1335b93b61666b433b

m1r0 commented on client/my-sites/domains/domain-management/list/style.scss:
```diff
@@ -327,4 +327,68 @@
+}
+
+// Similar to the sites-overview, hide the domains-overview list on the left on smaller screens.
+@media (max-width: 1280px) {
```
Same as the previous comment.

Imran92 replied:
Removing it in favor of this PR https://github.com/Automattic/wp-calypso/pull/97240

m1r0 commented:
Looks good overall! I have just a few small notes.

* There's some extra margin at the bottom of the list. It's visible when the list is scrollable.

<img width="419" alt="image" src="https://github.com/user-attachments/assets/9394ecb6-f6ed-4485-a320-9e7e02cd5a94">

* As a suggestion (could be done in another PR), WDYT of showing the meatballs menu (the 3 dots) on hover similar to `/sites`? To make it even more consistent, we could rotate the dots vertically.
* Another suggestion - it would be nice to have an active state for the selected domain in the sidebar (same as `/sites`).

Imran92 committed:
Use mixin for breakpoint

Imran92 committed:
Remove sidebar hiding style

Imran92 committed:
Fix table scroll and icon rotation

Imran92 committed:
Rotate icon and make visible on hover

Imran92 commented:
> There's some extra margin at the bottom of the list. It's visible when the list is scrollable.

I've removed the space [here](https://github.com/Automattic/wp-calypso/pull/97193/commits/6388b61133747689b6dbdcacbb8d7f5de5b5eb71). Have also changed the scroll behavior to scroll including the search bar, to match the behavior of /sites.

> As a suggestion (could be done in another PR), WDYT of showing the meatballs menu (the 3 dots) on hover similar to /sites? To make it even more consistent, we could rotate the dots vertically.

I've made the meatball menu vertical and show on hover as well.

> Another suggestion - it would be nice to have an active state for the selected domain in the sidebar (same as /sites).

Good suggestion, but I've intentionally not done that yet, because the selected site determination process will change significantly with DataViews. So to avoid re-works, I think it'll be better to do it once we have the DataViews

m1r0 reviewed (commented):
>  I've made the meatball menu vertical and show on hover as well.

Looks good!

> I've removed the space [here](https://github.com/Automattic/wp-calypso/pull/97193/commits/6388b61133747689b6dbdcacbb8d7f5de5b5eb71). Have also changed the scroll behavior to scroll including the search bar, to match the behavior of /sites.

This is fixed on big resolutions but I'm still seeing the white space below 1400px. 🤔 Any ideas why that might be?

https://github.com/user-attachments/assets/d6a700c4-3470-4434-9b84-381ede954711

On the same note, after the latest changes, I've noticed some white space around the sidebar items. It looks like the items are no longer full-width.

<img width="423" alt="image" src="https://github.com/user-attachments/assets/a6b71438-b36e-45c2-9d75-21fd1ecac3c8" />

> Good suggestion, but I've intentionally not done that yet, because the selected site determination process will change significantly with DataViews. So to avoid re-works, I think it'll be better to do it once we have the DataViews

Good to know, thanks!

Imran92 committed:
Fix extra spacing in smaller screensize

Imran92 commented:
> This is fixed on big resolutions but I'm still seeing the white space below 1400px. 🤔 Any ideas why that might be?

Thanks for finding it! I've fixed the issue here - https://github.com/Automattic/wp-calypso/pull/97193/commits/d5998ef3050396dbd84c7f70102d8e618e2ea9b8

> On the same note, after the latest changes, I've noticed some white space around the sidebar items. It looks like the items are no longer full-width.

Not sure how it's happening there, I could reproduce this spacing only below 960px. In that case, hopefully it won't be an issue because once the following PR gets merged, the domain list will be hidden below 1280px

https://github.com/Automattic/wp-calypso/blob/415cea642be3a6e565ff561f06bb7bb9dad6063c/client/my-sites/domains/domain-management/list/style.scss#L328-L335

m1r0 commented:
> Not sure how it's happening there, I could reproduce this spacing only below 960px.

Interesting. For me, it happens on all resolutions. I can also reproduce it on Firefox. I've tracked it down to [this change](https://github.com/Automattic/wp-calypso/pull/97193/commits/6388b61133747689b6dbdcacbb8d7f5de5b5eb71#diff-6c5972cdc95feb1074312c8ec2c57e557993e7797e0244bc6e08a51c08ac5582R240-R241) (flex on `.main`). A fix that works for me is using `width: 100%` to `.domains-table`.

Imran92 committed:
Fix scrollbar being visible in tabs

Imran92 commented:
> Interesting. For me, it happens on all resolutions (as seen in the video above). I can also reproduce it on Firefox. I've tracked it down to [this change](https://github.com/Automattic/wp-calypso/pull/97193/commits/6388b61133747689b6dbdcacbb8d7f5de5b5eb71#diff-6c5972cdc95feb1074312c8ec2c57e557993e7797e0244bc6e08a51c08ac5582R240-R241) (flex on .main). A fix that works for me is using width: 100% to .domains-table.

I still can't reproduce it after many tries 🤔 Can you kindly also try the calypso.live link above? https://github.com/Automattic/wp-calypso/pull/97193#issuecomment-2524189772

Imran92 commented:
I've also fixed the issue https://github.com/Automattic/wp-calypso/issues/97390 here. So the following scrollbar scenario should not happen -

![image](https://github.com/user-attachments/assets/5ab3a305-31c6-4c66-a363-4b740b11995a)

m1r0 commented:
> I still can't reproduce it after many tries 🤔 Can you kindly also try the calypso.live link above and see what happens?

It's even worse on calypso.live. The extra spacing is still there, but also the "Domains" header title is missing some spacing... 😞 I did a video in Firefox, but it happens in Chrome as well. Maybe ask someone else from the team to test it?

https://github.com/user-attachments/assets/70669eba-f322-46c5-b736-95434632d498

> I've also fixed the issue https://github.com/Automattic/wp-calypso/issues/97390 here. So the following scrollbar scenario should not happen -

I've [fixed](https://github.com/Automattic/wp-calypso/pull/97240/files#diff-8bb4f3f403bd6b260e3987bef17afc8f2318d3c0b702461dcb5c724e0b46adf9R40) the same thing. 😄 Let me know if I should remove it from my PR.

m1r0 committed:
Fix domains list not being full-width on some envs

m1r0 committed:
Fix sidebar heading padding on some envs

m1r0 committed:
Fix domain search padding on some envs

m1r0 reviewed (approved):
As we discussed, I fixed the issues that were popping up on my end and calypso.live.

@Imran92, could you give it a final look before merging?

Imran92 commented:
Thanks @m1r0 ! LGTM and merging
